Assets/Scripts/TankControllerAndroid.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TankControllerAndroid : MonoSingletonGeneric<TankControllerAndroid>
+{
+    [SerializeField]
+    private Joystick joystick;
+
+    [SerializeField]
+    private new Rigidbody rigidbody;
+
+    private float dirX, dirY, moveSpeed;
+
+    private void Start()
+    {
+        moveSpeed = 0.2f;
+    }
+
+    private void Update()
+    {
+        dirX = joystick.Horizontal * moveSpeed;
+        dirY = joystick.Vertical * moveSpeed;
+
+        rigidbody.transform.localPosition += Vector3.forward * dirY;
```

Comment:
Check here first if dirX or dirY is greater than zero then move to the next code like this 
if(dirX || dirY)
{
 rigidbody.transform.localPosition += Vector3.forward * dirY;
 rigidbody.transform.localPosition += Vector3.right * dirX;
}